aeron-archive/src/main/java/io/aeron/archive/client/ReplayMerge.java

```diff
@@ -506,6 +510,31 @@ else if (shouldStopAndRemoveReplay(position))
         return workCount;
     }
 
+    private boolean callGetMaxRecordedPosition(final long nowMs)
+    {
+        if (nowMs < timeOfNextGetMaxRecordedPositionMs)
+        {
+            return false;
+        }
+
+        final long correlationId = archive.context().aeron().nextCorrelationId();
+
+        final boolean result = archive.archiveProxy().getMaxRecordedPosition(
+            recordingId, correlationId, archive.controlSessionId());
+
+        if (result)
+        {
+            activeCorrelationId = correlationId;
+        }
+
+        // increase backoff regardless of result
+        getMaxRecordedPositionBackoffMs = Long.min(
+            getMaxRecordedPositionBackoffMs * 2, GET_MAX_RECORDED_POSITION_BACKOFF_MAX_MS);
+        timeOfNextGetMaxRecordedPositionMs = nowMs + getMaxRecordedPositionBackoffMs;
```

Comment:
This back-off reset can happen already when transitioning to `REPLAY` state.

There might be state flickering back and force between `CATCHUP` and `ATTEMPT_LIVE_JOIN`. The latter performs those excessive calls to resolve position which we want to throttle.

Reply:
I'm not sure what you're getting at here.  I reset the backoff when moving to `CATCHUP` the first time.  But after that, each time the state flickers back and forth, I let the backoff increase.  It was that flickering that was previously causing the vast numbers of calls to `getRecordingPosition`.

Reply:
Just wanted to mention that we can reset the back-off counter one state earlier. But this won't change the outcome so I'm fine with either approach.